File: src/main/java/fun/vs/fw/demo/kafka/KafkaListenerConfiguration.java
package fun.vs.fw.demo.kafka;


import fun.vs.fw.demo.domain.TransactionCategorizer;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.kafka.annotation.EnableKafka;

@Configuration
@EnableKafka
public class KafkaListenerConfiguration {

    @Bean
    public TransactionKafkaListener transactionKafkaListener(TransactionCategorizer categorizationService) {
        return new TransactionKafkaListener(categorizationService);
    }
}

File: fvf4k/src/main/kotlin/fvf4k/demo/infra/kafka/KafkaListenerConfiguration.kt
package fvf4k.demo.infra.kafka

import fvf4k.demo.domain.api.CategorizeTransaction
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.kafka.annotation.EnableKafka


@Configuration
@EnableKafka
class KafkaListenerConfiguration {

    @Bean
    fun transactionKafkaListener(categorizeTransaction: CategorizeTransaction) =
        TransactionKafkaListener(categorizeTransaction)
}